## Statistical Data Analysis

### Data Analysis

In [1]:
import pandas as pd
import chardet
import scipy
from scipy.stats import poisson

In [2]:
with open('data/premier_league_data_consolidated.tsv', 'rb') as f:
    result = chardet.detect(f.read())    

In [4]:
df =pd.read_csv('data/premier_league_data_consolidated.tsv',delimiter='\t',encoding = result['encoding'])

### Hypothesis Test

Average home goal score for Chelsea in 2015/16 season was 1.6842. They made some changes to strategy to do better in next season. In the first 6 games for Arsenal in next season which is 2016/17 they scored goals. Is there enough evidence at 5% significance level which suggest that game plan for new seasoned worked and goal scoring increased for Chelsea. 

** Probability of events in Poisson distribution **

$$P(k) = \frac{y^ke^{-y}}{k!}$$

In [5]:
df_s_1516 = df[df['seasonlabel']=='2015/16']

mean_gs_1516 = round(df_s_1516['score_h'].mean(),4)

mean_gs_1516_c = round(df_s_1516[df_s_1516['team_h']=='Chelsea']['score_h'].mean(),4)

print('Mean Home Goal Scored by Chelsea', mean_gs_1516_c)


Mean Home Goal Scored by Chelsea 1.6842


P(k) is probability of k event in interval 

** $H_{0}$ (Null Hypothesis) ** is Chelsea scoring rate at home  is equal to 1.684; ** k = 1.684 **

** $H_{1}$ (Alternate Hypothesis)** is Chelsea scoring rate at home  is greater than 1.684; ** k > 1.684 **

** Conduct Test ** 

In [7]:
n_match = 6

df_s_1617 = df[df['seasonlabel']=='2016/17']


df_1617_c = df_s_1617[df_s_1617['team_h']=='Chelsea'].sort_values('gameid').head(n_match)

gs_1617_c = df_1617_c['score_h'].sum()

mean_gs_1617_c = round(df_s_1617[df_s_1617['team_h']=='Chelsea']['score_h'].mean(),4)

test_value = gs_1617_c
lda= mean_gs_1516_c*n_match

print ('Test Value :',test_value)
print ('Rate or Lambda :',lda)

Test Value : 18
Rate or Lambda : 10.1052


In [10]:
def get_poisson_prob(gs_count) :
    poisson_prob = {}
    for i in range (gs_count):
        poisson_prob[i] = poisson.cdf(i,lda)
    return poisson_prob

In [11]:
poisson_prob = get_poisson_prob(25)
for k,v in poisson_prob.items():
    print (k,'\t',round(v,5))

0 	 4e-05
1 	 0.00045
2 	 0.00254
3 	 0.00957
4 	 0.02732
5 	 0.06321
6 	 0.12365
7 	 0.21089
8 	 0.3211
9 	 0.44484
10 	 0.56988
11 	 0.68475
12 	 0.78148
13 	 0.85667
14 	 0.91095
15 	 0.94751
16 	 0.9706
17 	 0.98433
18 	 0.99204
19 	 0.99613
20 	 0.9982
21 	 0.9992
22 	 0.99966
23 	 0.99986
24 	 0.99994


We need to find $P(k>=18)$ which will be equal to 1- $P(k<=17)$ 

From the table above $P(k<=17)$ is 0.98433

$P(k >=18)$ = 1 - 0.98433 = 0.0157 = 1.57%

** At 5% significance level there is enough evidence to suggest that, with same rate of scoring from 2015/16 season in home game there is no way Chelsea would have scored 18 goals in first 6 games of new season. This means the rate of scoring has increased. We will accept the alternate hypothesis and reject the null hypothesis**